### ISSUE TITLE
New bugs related to candidate for 3.0.0

### ISSUE THREAD

**Issue by staerz** [jeremiah-c-leary/vhdl-style-guide#487]
**Environment**
Latest version 2.0.0-457-g30bfcad7.

**Describe the bug**
This is a collection of bugs and issues found with the new version for new features.

**Expected behavior**

### :heavy_check_mark: Broken link to Configuring Type of Instantiations
The doc for `instantiation_034` points to 'Configuring Type of Instantiations' but the link is broken, although the page actually exists (dashes vs. underscores)

### :heavy_check_mark: Unclear error message upon unexpected error
If the parsing fails entirely, then the following error is thrown:
```bash
Error: Unknown token in component_instantiation_statement(275:3)
       Expecting ;, found port
```
It doesn't give any hint where the mistake might be.
A line number and file name would be helpful to be added to that error.

### :heavy_check_mark: instantiation_003 circular error message
The message has a wrong second reference.
```
  instantiation_003         | Error      |        358 | Add 1 space(s) between inst_sctrl_tester and inst_sctrl_tester
```
Instead the message should say: `Add 1 space(s) between inst_sctrl_tester and the colon`.

### :heavy_check_mark: Typo in doc for [instantiation_010](https://vhdl-style-guide.readthedocs.io/en/latest/instantiation_rules.html#instantiation-010)
The keyword `Violation` should be on the next line.

### :heavy_check_mark: concurrent_004 circular error message
```
  concurrent_004            | Error      |        757 | Add 1 space(s) between ) and )
```
The second character is a `<` in fact:
```vhdl
    user_led_r(2) <= not clk_feb_locked;
```
So it looks like the second argument is wrong ...

**and it seems that this behaviour is true for all rules that check the spaces, i.e. also the new `block_100`**

### :heavy_check_mark: Typo in block_202 documentation
- In the examples for violation and fix and underscore is missing in the `block label` label.
- In the rule description, there are two trailing dots.

### :heavy_check_mark: Configuration of blank lines (#504)
Some rules on blank lines allow to "allow comments".
That is usually a good idea, but for the "check above" rules, if that's configured, a comment "overrides" the rule of requesting a blank line: Let's take the example [`block_200`](https://vhdl-style-guide.readthedocs.io/en/latest/block_rules.html#block-200). With "allow_comment" active, the following code passes:

```vhdl
a <= b;
-- some comment
block_label : block is
```

While the intended check should be
```vhdl
a <= b;

-- some comment
-- and maybe that comment goes
-- even over multiple lines
block_label : block is
```

### :heavy_check_mark: Documentation for [`architecture-002`](https://vhdl-style-guide.readthedocs.io/en/latest/architecture_rules.html#architecture-002)
The rule has been split. It would nice if these rules would be a link to access them quicker.

### :heavy_check_mark: Add a rule for `architecture` similar to `block_205`
A file doesn't necessarily end after the `end architecture` as another architecture, a package or what ever could be defined in the same file.
Hence adding such a rule (that would not enforce a blank line if it were at file end, of course) would be useful.

### :heavy_check_mark: Update copyright statement
Currently the copyright states 2017-2018 ... I think it should be updated to 2017-2020 (and the second year should likely at best be derived by `date`).

### :heavy_check_mark: Installation warning
When installing via `python setup.py install` the following warning is thrown:
```
warning: no files found matching 'README.md'
```
It seems to stem from an invalid installation directive.

... I'll keep adding things along the way of testing ...

**Comment by jeremiah-c-leary** [comment]
@staerz ,
I believe I fixed the link.

**Comment by staerz** [comment]
Yup, the link now works.

I have added a few other issues though. I'm done checking for today and will likely be able to pursue the next couple of days, so let me know if you need any feedback.

**Comment by jeremiah-c-leary** [comment]
> Configuration of blank lines

Just to clarify the issue...you would like the comments to be considered part of the rule check.  Such that a blank line would be added above the comments?

**Comment by staerz** [comment]
> > Configuration of blank lines
> 
> Just to clarify the issue...you would like the comments to be considered part of the rule check. Such that a blank line would be added above the comments?

Yes, blank line above the comment (required). If there is no comment, also no problem, as that falls back to the current situation.

**Comment by jeremiah-c-leary** [comment]
Hey @staerz ,
I believe the following issues have been addressed:
* Unclear error message upon unexpected error
* instantiation_003 circular error message
* Typo in doc for instantiation_010
* Documentation for architecture-002
* Update copyright statement

**Comment by staerz** [comment]
Alright, I updated the issue description. In fact you also already fixed the other typo doc (block_202)

**Comment by jeremiah-c-leary** [comment]
Hey @staerz ,

I put in a rule to handle this issue:
* Add a rule for architecture similar to block_205

When you get a chance could you validate it.

Thanks,

--Jeremy

**Comment by staerz** [comment]
Hi @jeremiah-c-leary,

I have reverted to the status d7403e62 ("Fixing 3.0.0 release date") and got the following bug, when `comment_001` is disabled:

```
Traceback (most recent call last):
  File "/usr/local/bin/vsg", line 11, in <module>
    load_entry_point('vsg===master', 'console_scripts', 'vsg')()
  File "/usr/local/lib/python3.8/site-packages/vsg-master-py3.8.egg/vsg/__main__.py", line 418, in main
    oRules.check_rules(bAllPhases=commandLineArguments.all_phases, lSkipPhase=commandLineArguments.skip_phase)
  File "/usr/local/lib/python3.8/site-packages/vsg-master-py3.8.egg/vsg/rule_list.py", line 215, in check_rules
    oRule.analyze(self.oVhdlFile)
  File "/usr/local/lib/python3.8/site-packages/vsg-master-py3.8.egg/vsg/rules/align_tokens_in_region_between_tokens_when_between_tokens_unless_between_tokens.py", line 63, in analyze
    lToi = oFile.get_tokens_bounded_by_token_when_between_tokens(self.left_token, self.right_token, self.lBetween[0], self.lBetween[1])
  File "/usr/local/lib/python3.8/site-packages/vsg-master-py3.8.egg/vsg/vhdlFile/vhdlFile.py", line 164, in get_tokens_bounded_by_token_when_between_tokens
    return extract.get_tokens_bounded_by_token_when_between_tokens(oLeft, oRight, oStart, oEnd, self.lAllObjects, self.oTokenMap, include_trailing_whitespace)
  File "/usr/local/lib/python3.8/site-packages/vsg-master-py3.8.egg/vsg/vhdlFile/extract/get_tokens_bounded_by_token_when_between_tokens.py", line 14, in get_tokens_bounded_by_token_when_between_tokens
    if iStart < iLeft and iRight < iEnd:
TypeError: '<' not supported between instances of 'NoneType' and 'int'
```

I traced it down to the commit 712513d2 (related to #501) which seems to introduce this bug: I removed that commit from the history and ran again, then it's fine.

I also hereby confirm that the commits related to introducing `architecture_200` work fine, I'll update the issue description.

**Comment by jeremiah-c-leary** [comment]
Hey @staerz ,

I was reviewing this issue and it looks like there are three remaining items:

1) Configuration of blank lines
2) Installation warning
3)  comment_001

I believe the first one is covered by issue #504

I tried the second one and installed without an error.

For the third one, comment_001 no longer exists.  Looking at the log file it appears to have occured back in March.  Could you check if this is still an issue?

Thanks,

--Jeremy

**Comment by staerz** [comment]
Hi @jeremiah-c-leary,

I updated the issue description, yes, 1) is covered by #504.

2) Is still occurring. This seems to be an issue with your `MANIFEST.in`: It contains the line `include README.md`, but there is no `README.md` in the repository.

3) `comment_001`: I think I never mentioned it, so sure, that's done.

**Comment by jeremiah-c-leary** [comment]
Hey @staerz ,

I changed README.md to README.rst in the MANIFEST.in file.  I tested it and I no longer see the warning.

**Comment by staerz** [comment]
Happy you found the bug.

Since that now addresses the last open point, please feel free to close this issue :+1:

**Comment by jeremiah-c-leary** [comment]
Woot!!
